lib/spack/spack/binary_distribution.py

```diff
@@ -1401,32 +1401,44 @@ def nodes_to_be_packaged(specs, root=True, dependencies=True):
         return list(filter(packageable, nodes))
 
 
-def push(specs, push_url, include_root: bool = True, include_dependencies: bool = True, **kwargs):
-    """Create a binary package for each of the specs passed as input and push them
-    to a given push URL.
+def push(
+    spec: Spec,
+    mirror_url,
+    force=False,
+    relative=False,
+    unsigned=False,
+    allow_root=False,
+    key=None,
+    regenerate_index=False,
+):
+    """Create and push binary package for a single spec to the specified
+    mirror url.
 
     Args:
-        specs (List[spack.spec.Spec]): installed specs to be packaged
-        push_url (str): url where to push the binary package
-        include_root (bool): include the root of each spec in the nodes
-        include_dependencies (bool): include the dependencies of each
-            spec in the nodes
-        **kwargs: TODO
+        spec (spack.spec.Spec): Spec to package and push
+        mirror_url (str): Desired destination url for binary package
+        kwargs: Passed through to safe_push(...)
 
-    """
-    # Be explicit about the arugment type
-    if type(include_root) != bool or type(include_dependencies) != bool:
-        raise ValueError("Expected include_root/include_dependencies to be True/False")
+    Returns:
+        True if package was pushed, False otherwise.
 
-    nodes = nodes_to_be_packaged(specs, root=include_root, dependencies=include_dependencies)
+    """
+    try:
+        safe_push(
+            spec,
+            mirror_url,
+            force=force,
+            relative=relative,
+            unsigned=unsigned,
+            allow_root=allow_root,
+            key=key,
+            regenerate_index=regenerate_index,
+        )
+    except NoOverwriteException as e:
+        warnings.warn(str(e))
+        return False
```

Comment:
:100: To this. This function does one thing: error handling.

Reply:
Meh, it handles one particular error, it doesn't handle upload errors or w/e. It's just there cause it did this in the past.

Reply:
True, but in cmd/buildcache.py there's it calls safe_push directly so that in principle any error can be handled, and getting True/False with a logged warning is strictly less useful :p